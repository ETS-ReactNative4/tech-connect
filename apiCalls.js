export const getLocations = async () => {
    const url = 'https://tech-connect-be.herokuapp.com/api/v1/locations'
    const response = await fetch(url)
    const locationData = await response.json()
    return locationData.data.map(location => {
      return {label: location.attributes.city}
    })
  }

export const getEmployers = async () => {
  const url = 'https://tech-connect-be.herokuapp.com/api/v1/employers'
  const response = await fetch(url)
  const employerData = await response.json()
  return employerData.data.map(location => {
    return {label: location.attributes.name}
  })
}


export const getPositions = async () => {
  const url = 'https://tech-connect-be.herokuapp.com/api/v1/positions'
  const response = await fetch(url)
  const positionData = await response.json()
  return positionData.data.map(location => {
    return {label: location.attributes.job_title}
  })
}

export const getAllUsers = async (key) => {
  const url = `https://tech-connect-be.herokuapp.com/api/v1/users?api_key=${key}`
  const response = await fetch(url)
  const userData = await response.json()
  return userData.data.map(user => {
    return {...user.attributes, id: user.id}
  })
}

export const getUsersFilter = async (key, filter, name) => {
  const url = `https://tech-connect-be.herokuapp.com/api/v1/users/find?${filter}=${name}&api_key=${key}`
  const response = await fetch(url)
  const userData = await response.json()
  return userData.data.map(user => {
    return {...user.attributes, id: user.id}
  })
}

export const getUserInfo = async (id, apiKey) => {

  const url = `https://tech-connect-be.herokuapp.com/api/v1/users/${id}?api_key=${apiKey}`
  try {
    const response = await fetch(url)
    if (!response.ok) {
      const errorText = await response.json()
      throw Error(errorText.error)
    }
    
    const userInfo = await response.json()
    const cleanedUser = {id: userInfo.data.id, ...userInfo.data.attributes}
    return cleanedUser
  } catch(err) {
    console.log(err)
  }
}