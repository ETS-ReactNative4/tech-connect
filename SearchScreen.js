import React from 'react';
import { StyleSheet, Text, View, Button, TouchableHighlight, Image, ScrollView, ActivityIndicator } from 'react-native';
import { connect } from 'react-redux';
import { SearchBar, ButtonGroup } from 'react-native-elements'
import { LinearGradient } from 'expo'
import SuggestedConnection from './SuggestedConnection.js' 
import Connection from './Connection'
import { getAllUsers, getUserInfo, getUsersFilter } from './apiCalls'
import Icon from 'react-native-vector-icons/Feather'


export class SearchScreen extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      allUsers: [],
      search: '',
      selectedIndex: 0,
      loading: true
    }
  }

  async componentDidMount() {
    const allUsers = await getAllUsers(this.props.user.api_key)
    this.setState({ allUsers, loading: false })
  }

  viewProfile = async (id) => {
    const user = await getUserInfo(id, this.props.user.api_key)
    this.props.navigation.navigate('ProfilePage', {user})
  }

  updateIndex = (selectedIndex) => {
    this.setState({selectedIndex})
  }

  getFilteredUsers = async () => {
    this.setState({ allUsers: [], loading: true })
    const { selectedIndex, search } = this.state
    const queryParams = ['name', 'city', 'position', 'employer']
    const allUsers = await getUsersFilter(this.props.user.api_key, queryParams[selectedIndex], search)
    this.setState({ allUsers, loading: false })
  }

  render() {
    const { allUsers, selectedIndex, search, loading } = this.state
    const userArray = allUsers.map(user => <Connection connection={ user } viewProfile={(() => this.viewProfile(user.id)) } key={ user.id }/>)
    const buttons = ['Name', 'Location', 'Position', 'Employer']

    return (
      <View style={styles.container}>
      <View style={styles.componentContainer}>
        <Text style={styles.search}>Search by:</Text>
        <ButtonGroup
          onPress={this.updateIndex}
          selectedIndex={selectedIndex}
          selectedButtonStyle={{backgroundColor: '#4AA9C5'}}
          buttons={buttons}
          containerStyle={{height: 30}}
        />
        <SearchBar
          onSubmitEditing={ this.getFilteredUsers }
          searchIcon= {
            <LinearGradient
              colors={['#4AA9C5', '#6364c1', '#93548F']}
              start={0.4}
              style={styles.gradient} >
              <Icon
                name='search'
                size={16}
                color='#FFF'
                onPress={ this.getFilteredUsers }
              />
            </LinearGradient>
          }
          leftIconContainerStyle={styles.icon} 
          containerStyle={styles.searchContainer}
          inputContainerStyle={styles.inputContainer}
          lightTheme
          placeholder="Type Here..."
          onChangeText={(text) => this.setState({search: text})} 
          value={search}
        />
        <ScrollView contentContainerStyle={{alignItems: 'center'}} style={ styles.scrollContainer }>
              <View style={ styles.connectionsContainer }>
                { !loading ? userArray : <ActivityIndicator size="large" color="#fff" /> }
                { 
                  (!userArray.length && !loading) && 
                    <Text style={{color: '#fff'}}>No users were found.</Text> 
                }
              </View>
        </ScrollView>
      </View>
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(SearchScreen)


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'stretch',
    justifyContent: 'center',
    height: '100%',
    width: '100%',
    paddingTop: 20,
    backgroundColor: '#4AA9C5'
  },
  search: {
    marginTop: 10, 
    marginLeft: 10,
    fontSize: 20
  },
  icon: {
    marginRight: 5,
    marginLeft: 3,
    width: 36,
    height: 36,
    borderRadius: 50,
    backgroundColor: '#4AA9C5'
  },
  gradient: {
    borderRadius: 300, 
    width: 36, 
    height: 36, 
    justifyContent: 'center', 
    alignItems: 'center'
  },
  componentContainer: {
    flex: 1,
    backgroundColor: '#fff', 
    borderRadius: 5,
    justifyContent: 'center', 
    padding: 15,
    alignItems: 'stretch',
    marginTop: 25,
    marginRight: 20,
    marginLeft: 20,
  },
  searchContainer: {
    width: '100%',
    marginBottom: 10, 
    backgroundColor: '#ffffff00', 
    borderBottomWidth: 0, 
    borderTopWidth: 0
  },
  inputContainer: {
    borderRadius: 30, 
    borderWidth: 2, 
    borderColor: '#4AA9C5',
    borderBottomWidth: 2, 
    backgroundColor: '#fff'
  },
  scrollContainer: {
    flex: 1,
  },
  innerContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
    width: '100%',
    backgroundColor: '#FFF',
    paddingBottom: 20,
  },
  connectionsContainer: {
    backgroundColor: '#4AA9C5',
    padding: 10,
    borderRadius: 5,
    width: '100%'
  }
})




