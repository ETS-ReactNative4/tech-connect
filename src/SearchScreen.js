import React from 'react';
import { StyleSheet, Text, TextInput, View, Button, TouchableHighlight, Image, ScrollView, ActivityIndicator } from 'react-native';
import { connect } from 'react-redux';
import { ButtonGroup } from 'react-native-elements'
import { LinearGradient } from 'expo'
import Connection from './Connection'
import { getAllUsers, getUserInfo, getUsersFilter } from '../apiCalls'
import Icon from 'react-native-vector-icons/Feather'


export class SearchScreen extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      allUsers: [],
      search: '',
      selectedIndex: 0,
      loading: true
    }
  }

  async componentDidMount() {
    const allUsers = await getAllUsers(this.props.user.api_key)
    this.setState({ allUsers, loading: false })
  }

  viewProfile = async (id) => {
    const user = await getUserInfo(id, this.props.user.api_key)
    this.props.navigation.navigate('ProfilePage', {user})
  }

  updateIndex = (selectedIndex) => {
    this.setState({selectedIndex})
  }

  getFilteredUsers = async () => {
    this.setState({ allUsers: [], loading: true })
    const { selectedIndex, search } = this.state
    const queryParams = ['name', 'city', 'position', 'employer']
    const allUsers = await getUsersFilter(this.props.user.api_key, queryParams[selectedIndex], search)
    this.setState({ allUsers, loading: false })
  }

  render() {
    const { allUsers, selectedIndex, search, loading } = this.state
    const userArray = allUsers.map(user => <Connection connection={ user } viewProfile={this.viewProfile.bind(null,user.id)} key={ user.id }/>)
    const buttons = ['Name', 'Location', 'Position', 'Employer']

    return (
      <View style={styles.container}>
        <View style={styles.componentContainer}>
          <Text style={styles.search}>Search by:</Text>
          <ButtonGroup
            onPress={this.updateIndex}
            selectedIndex={selectedIndex}
            selectedButtonStyle={{backgroundColor: '#4AA9C5'}}
            buttons={buttons}
            containerStyle={{height: 30}}
          />
          <View style={styles.searchContainer}>
            <LinearGradient
              colors={['#4AA9C5', '#6364c1', '#93548F']}
              start={0.4}
              style={styles.gradient} >
              <Icon
                name='search'
                size={16}
                color='#FFF'
                onPress={ this.getFilteredUsers }
              />
            </LinearGradient>
            <TextInput
              style={styles.inputContainer}
              placeholder="Search here..."
              underlineColorAndroid="transparent"
              onChangeText={(text) => this.setState({search: text})} 
              value={search}
            />
          </View>
          <ScrollView contentContainerStyle={{alignItems: 'center'}} style={ styles.scrollContainer }>
            <View style={ styles.connectionsContainer }>
              { !loading ? userArray : <ActivityIndicator size="large" color="#fff" /> }
              { 
                (!userArray.length && !loading) && 
                  <Text style={{color: '#fff'}}>No users were found.</Text> 
              }
            </View>
          </ScrollView>
        </View>
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(SearchScreen)


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'stretch',
    justifyContent: 'center',
    height: '100%',
    width: '100%',
    paddingTop: 20,
    backgroundColor: '#4AA9C5'
  },
  search: {
    marginTop: 10, 
    marginLeft: 10,
    fontSize: 20
  },
  gradient: {
    marginLeft: 4,
    borderRadius: 300, 
    width: 37, 
    height: 37, 
    justifyContent: 'center', 
    alignItems: 'center'
  },
  componentContainer: {
    flex: 1,
    backgroundColor: '#fff', 
    borderRadius: 5,
    justifyContent: 'center', 
    padding: 15,
    alignItems: 'stretch',
    marginTop: 25,
    marginRight: 20,
    marginLeft: 20,
  },
  searchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10, 
    borderRadius: 30, 
    borderWidth: 2, 
    borderColor: '#4AA9C5',
    backgroundColor: '#ffffff00', 
    borderBottomWidth: 2, 
  },
  inputContainer: {
    flex: 1,
    fontSize: 16,
    paddingTop: 12,
    paddingBottom: 12,
    paddingLeft: 10,
    backgroundColor: 'transparent'
  },
  scrollContainer: {
    flex: 1,
  },
  innerContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
    width: '100%',
    backgroundColor: '#FFF',
    paddingBottom: 20,
  },
  connectionsContainer: {
    backgroundColor: '#4AA9C5',
    padding: 10,
    borderRadius: 5,
    width: '100%'
  }
})
