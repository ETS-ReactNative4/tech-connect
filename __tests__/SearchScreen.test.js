import React from 'react'
import { SearchScreen, mapStateToProps } from '../SearchScreen'
import { shallow } from 'enzyme'
import { Input } from 'react-native-elements'
import { SearchBar, ButtonGroup } from 'react-native-elements'
import { getAllUsers, getUsersFilter } from '../apiCalls'
import { ActivityIndicator, Text } from 'react-native';
import Icon from 'react-native-vector-icons/Feather'



const mockUsers = [{name: 'Kaylee'}, {name: 'Kylie'}]
jest.mock('../apiCalls')
getAllUsers.mockImplementation(() => mockUsers)
getUsersFilter.mockImplementation(() => mockUsers)


import renderer from 'react-test-renderer'


describe('SearchScreen', () => {
  let wrapper
  let onPressEvent
  let navigation

  beforeEach(() => {
    navigation = { navigate: jest.fn() };
    onPressEvent = jest.fn()
    wrapper = shallow(<SearchScreen user={{api_key: 1234}} updateUser={jest.fn()} navigation={navigation} />)
  })

  it('renders the snapshot', () => {
    const tree = renderer.create(
      <SearchScreen />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should get all users when loaded and set it in state', () => {
    wrapper.instance().componentDidMount()
    expect(getAllUsers).toHaveBeenCalled()
  })

  it('should update the selected index when a button in the group is pressed', () => {
    const spy = jest.spyOn(wrapper.instance(), 'updateIndex')
    wrapper.instance().forceUpdate()
    wrapper.find(ButtonGroup).simulate('press')
    expect(spy).toHaveBeenCalled()
  })

  it('should set the search string in state on change of the input', () => {
    wrapper.find(SearchBar).simulate('changeText', 'Backend')
    wrapper.instance().forceUpdate()
    expect(wrapper.instance().state.search).toEqual('Backend')
  })

  it('should display the activity indicator if the results are loading', () => {
    expect(wrapper.find(ActivityIndicator).length).toBe(1)
  })

  it('should get the user info when viewProfile is called', () => {
    wrapper.instance().viewProfile()
    expect(getAllUsers).toHaveBeenCalled()
  })

  it('should navigate to the Profile page for the user when clicked', async () => {
    await wrapper.instance().viewProfile()
    expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled()
  })

  it('should set the state of allUsers to empty and loading to true on getFilteredUsers', async () => {
    wrapper.instance().getFilteredUsers()
    expect(wrapper.instance().state.loading).toEqual(true)
    expect(wrapper.instance().state.allUsers).toEqual([])
    await getUsersFilter()
    expect(wrapper.instance().state.allUsers).toEqual(mockUsers)
  })

  it('should set the state of allUsers to empty and loading to true on getFilteredUsers', () => {
    wrapper.instance().getFilteredUsers()
    expect(wrapper.instance().state.loading).toEqual(true)
    expect(wrapper.instance().state.allUsers).toEqual([])
  })

  it('should call getUsersFilter on search', () => {
    wrapper.instance().getFilteredUsers()
    expect(getUsersFilter).toHaveBeenCalled()
  })

   it('should display no users found if there are no users', () => {
    wrapper.instance().setState({ loading: false, allUsers: []})
    expect(wrapper.find(Text).at(1).props().children).toEqual('No users were found.')
  })

  describe('mapStateToProps', () => {
    it('should return a user object', () => {
      const mockState = {
        user: { name: 'Kylie' },
        connections: { name: 'Bob' },
        isLoading: false,
        error: ''
      }
      const expected = { user: mockState.user }
      const result = mapStateToProps(mockState)
      expect(result).toEqual(expected)
    })
  })
})
