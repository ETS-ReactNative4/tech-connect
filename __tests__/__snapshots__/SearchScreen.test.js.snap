// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScreen renders the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#4AA9C5",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 25,
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "marginLeft": 10,
          "marginTop": 10,
        }
      }
    >
      Search by:
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#e3e3e3",
          "borderRadius": 3,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 30,
          "marginBottom": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 5,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomLeftRadius": 3,
            "borderRightColor": "#bdc6cf",
            "borderRightWidth": 0,
            "borderTopLeftRadius": 3,
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="buttonGroupItem"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4AA9C5",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                }
              }
              testID="buttonGroupItemText"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Name
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderLeftColor": "#bdc6cf",
            "borderLeftWidth": 1,
            "borderRightColor": "#bdc6cf",
            "borderRightWidth": 1,
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="buttonGroupItem"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#5e6977",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                }
              }
              testID="buttonGroupItemText"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Location
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderRightColor": "#bdc6cf",
            "borderRightWidth": 1,
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="buttonGroupItem"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#5e6977",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                }
              }
              testID="buttonGroupItemText"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Position
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomRightRadius": 3,
            "borderTopRightRadius": 3,
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="buttonGroupItem"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#5e6977",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                }
              }
              testID="buttonGroupItemText"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Employer
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff00",
          "borderBottomColor": "#e1e1e1",
          "borderBottomWidth": 0,
          "borderTopColor": "#e1e1e1",
          "borderTopWidth": 0,
          "marginBottom": 10,
          "padding": 8,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomWidth": 2,
              "borderColor": "#4AA9C5",
              "borderRadius": 30,
              "borderWidth": 2,
              "flexDirection": "row",
              "minHeight": 30,
              "overflow": "hidden",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4AA9C5",
                "borderRadius": 50,
                "height": 36,
                "justifyContent": "center",
                "marginLeft": 3,
                "marginRight": 5,
                "width": 36,
              }
            }
          >
            <ExponentLinearGradient
              colors={
                Array [
                  4283083205,
                  4284703937,
                  4287845519,
                ]
              }
              startPoint={
                Array [
                  undefined,
                  undefined,
                ]
              }
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 300,
                  "height": 36,
                  "justifyContent": "center",
                  "width": 36,
                }
              }
            >
              <Text />
            </ExponentLinearGradient>
          </View>
          <TextInput
            allowFontScaling={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onClear={[Function]}
            onFocus={[Function]}
            placeholder="Type Here..."
            placeholderTextColor="#86939e"
            platform="default"
            style={
              Object {
                "alignSelf": "center",
                "color": "#86939e",
                "flex": 1,
                "fontSize": 18,
                "marginLeft": 10,
                "minHeight": 40,
              }
            }
            testID="searchInput"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 15,
                "marginRight": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
